becca-new-app/src/App.js

```diff
@@ -2,29 +2,57 @@ import React, { useEffect, useState } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-const useSemiPersistentState = () => {
-	const initialTodoList =
-		JSON.parse(localStorage.getItem("savedTodoList")) || [];
-	const [todoList, setTodoList] = useState(initialTodoList);
+function App() {
+	const [todoList, setTodoList] = useState([]);
+	const [isLoading, setIsLoading] = useState(true);
 
 	useEffect(() => {
-		localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-	}, [todoList]);
+		const fetchData = async () => {
+			try {
+				const dataPromise = new Promise((resolve, reject) => {
+					setTimeout(() => {
+						resolve({ data: { todoList: [] } });
+					}, 2000);
+				});
+
+				dataPromise.then((result) => {
+					const { todoList: fetchedTodoList } = result.data;
+					setTodoList(fetchedTodoList);
+					setIsLoading(false);
+				});
+			} catch (error) {
+				setIsLoading(false);
+			}
+		};
+
+		fetchData();
+	}, []);
+	useEffect(() => {
```

Comment:
why is this useEffect here? The purpose of removing the loading is because the data is already fetched. This useEffect is only setting the data after you set it as loading.

I don't think we need this. Try to incorporate `localStorage.setItem("savedTodoList", JSON.stringify(todoList));` inside your `fetchData()` function